libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2412,6 +2412,29 @@ string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const
 			.render();
 		}
 
+		if (_from.category() == Type::Category::Array && _to.category() == Type::Category::Array)
+		{
+			auto const& fromArrayType = dynamic_cast<ArrayType const&>(_from);
+			auto const& toArrayType = dynamic_cast<ArrayType const&>(_to);
+
+			solAssert(!fromArrayType.baseType()->isDynamicallyEncoded(), "");
+			solAssert(fromArrayType.isByteArray() && toArrayType.isByteArray(), "");
+			solAssert(toArrayType.location() == DataLocation::Memory, "");
+			solAssert(fromArrayType.location() == DataLocation::CallData, "");
+			solAssert(toArrayType.isDynamicallySized(), "");
+
+			Whiskers templ(R"(
+				function <functionName>(offset, length) -> converted {
+					converted := <allocateMemoryArray>(length)
+					<copyToMemory>(offset, add(converted, 0x20), length)
+				}
+			)");
+			templ("functionName", functionName);
+			templ("allocateMemoryArray", allocateMemoryArrayFunction(toArrayType));
+			templ("copyToMemory", copyToMemoryFunction(fromArrayType.location() == DataLocation::CallData));
```

Comment:
it should be an assertion that from is calldata - otherwise there is nothing to convert.

Reply:
There is assertion just before constructing Whiskers, I think it should be fine.

Reply:
ok!